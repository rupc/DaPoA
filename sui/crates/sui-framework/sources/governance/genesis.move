// Copyright (c) Mysten Labs, Inc.
// SPDX-License-Identifier: Apache-2.0

module sui::genesis {
    use std::vector;

    use sui::balance;
    use sui::coin;
    use sui::clock;
    use sui::sui;
    use sui::sui_system;
    use sui::tx_context::TxContext;
    use sui::validator;
    use std::option;

    /// Initial value of the lower-bound on the amount of stake required to become a validator.
    /// TODO: testnet only. Needs to be changed.
    const INIT_MIN_VALIDATOR_STAKE: u64 = 1;

    /// Initial value of the upper-bound on the number of validators.
    const INIT_MAX_VALIDATOR_COUNT: u64 = 100;

    /// Stake subisidy to be given out in the very first epoch. Placeholder value.
    const INIT_STAKE_SUBSIDY_AMOUNT: u64 = 1000000;

    /// The initial balance of the Subsidy fund in Mist (1 Billion * 10^9)
    const INIT_STAKE_SUBSIDY_FUND_BALANCE: u64 = 1_000_000_000_000_000_000;

    /// This function will be explicitly called once at genesis.
    /// It will create a singleton SuiSystemState object, which contains
    /// all the information we need in the system.
    fun create(
        initial_sui_custody_account_address: address,
        validator_pubkeys: vector<vector<u8>>,
        validator_network_pubkeys: vector<vector<u8>>,
        validator_worker_pubkeys: vector<vector<u8>>,
        validator_proof_of_possessions: vector<vector<u8>>,
        validator_sui_addresses: vector<address>,
        validator_names: vector<vector<u8>>,
        validator_descriptions: vector<vector<u8>>,
        validator_image_urls: vector<vector<u8>>,
        validator_project_urls: vector<vector<u8>>,
        validator_net_addresses: vector<vector<u8>>,
        validator_p2p_addresses: vector<vector<u8>>,
        validator_consensus_addresses: vector<vector<u8>>,
        validator_worker_addresses: vector<vector<u8>>,
        validator_gas_prices: vector<u64>,
        validator_commission_rates: vector<u64>,
        protocol_version: u64,
        epoch_start_timestamp_ms: u64,
        ctx: &mut TxContext,
    ) {
        let sui_supply = sui::new(ctx);
        let subsidy_fund = balance::split(&mut sui_supply, INIT_STAKE_SUBSIDY_FUND_BALANCE);
        let storage_fund = balance::zero();
        let validators = vector::empty();
        let count = vector::length(&validator_pubkeys);
        assert!(
            vector::length(&validator_sui_addresses) == count
                && vector::length(&validator_names) == count
                && vector::length(&validator_descriptions) == count
                && vector::length(&validator_image_urls) == count
                && vector::length(&validator_project_urls) == count
                && vector::length(&validator_net_addresses) == count
                && vector::length(&validator_p2p_addresses) == count
                && vector::length(&validator_consensus_addresses) == count
                && vector::length(&validator_worker_addresses) == count
                && vector::length(&validator_gas_prices) == count
                && vector::length(&validator_commission_rates) == count,
            1
        );
        let i = 0;
        while (i < count) {
            let sui_address = *vector::borrow(&validator_sui_addresses, i);
            let pubkey = *vector::borrow(&validator_pubkeys, i);
            let network_pubkey = *vector::borrow(&validator_network_pubkeys, i);
            let worker_pubkey = *vector::borrow(&validator_worker_pubkeys, i);
            let proof_of_possession = *vector::borrow(&validator_proof_of_possessions, i);
            let name = *vector::borrow(&validator_names, i);
            let description = *vector::borrow(&validator_descriptions, i);
            let image_url = *vector::borrow(&validator_image_urls, i);
            let project_url = *vector::borrow(&validator_project_urls, i);
            let net_address = *vector::borrow(&validator_net_addresses, i);
            let p2p_address = *vector::borrow(&validator_p2p_addresses, i);
            let consensus_address = *vector::borrow(&validator_consensus_addresses, i);
            let worker_address = *vector::borrow(&validator_worker_addresses, i);
            let gas_price = *vector::borrow(&validator_gas_prices, i);
            let commission_rate = *vector::borrow(&validator_commission_rates, i);
            vector::push_back(&mut validators, validator::new(
                sui_address,
                pubkey,
                network_pubkey,
                worker_pubkey,
                proof_of_possession,
                name,
                description,
                image_url,
                project_url,
                net_address,
                p2p_address,
                consensus_address,
                worker_address,
                // TODO Figure out if we want to instead initialize validators with 0 stake.
                // Initialize all validators with 1 Mist stake.
                balance::split(&mut sui_supply, 1),
                option::none(),
                gas_price,
                commission_rate,
                0, // start operating right away at epoch 0
                ctx
            ));
            i = i + 1;
        };

        sui_system::create(
            validators,
            subsidy_fund,
            storage_fund,
            INIT_MAX_VALIDATOR_COUNT,
            INIT_MIN_VALIDATOR_STAKE,
            INIT_STAKE_SUBSIDY_AMOUNT,
            protocol_version,
            epoch_start_timestamp_ms,
            ctx,
        );

        clock::create();

        // Transfer the remaining balance of sui's supply to the initial account
        sui::transfer(coin::from_balance(sui_supply, ctx), initial_sui_custody_account_address);
    }
}
