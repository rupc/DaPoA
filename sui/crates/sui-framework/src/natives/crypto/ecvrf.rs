// Copyright (c) Mysten Labs, Inc.
// SPDX-License-Identifier: Apache-2.0
use crate::legacy_empty_cost;
use fastcrypto::vrf::ecvrf::ECVRFProof;
use fastcrypto::vrf::VRFProof;
use move_binary_format::errors::PartialVMResult;
use move_vm_runtime::native_functions::NativeContext;
use move_vm_types::{
    loaded_data::runtime_types::Type,
    natives::function::NativeResult,
    pop_arg,
    values::{Value, VectorRef},
};
use smallvec::smallvec;
use std::collections::VecDeque;

pub const INVALID_ECVRF_HASH_LENGTH: u64 = 1;
pub const INVALID_ECVRF_PUBLIC_KEY: u64 = 2;
pub const INVALID_ECVRF_PROOF: u64 = 3;

pub fn ecvrf_verify(
    _context: &mut NativeContext,
    ty_args: Vec<Type>,
    mut args: VecDeque<Value>,
) -> PartialVMResult<NativeResult> {
    debug_assert!(ty_args.is_empty());
    debug_assert!(args.len() == 4);

    let proof_bytes = pop_arg!(args, VectorRef);
    let public_key_bytes = pop_arg!(args, VectorRef);
    let alpha_string = pop_arg!(args, VectorRef);
    let hash_bytes = pop_arg!(args, VectorRef);

    // TODO: implement native gas cost estimation https://github.com/MystenLabs/sui/issues/3593
    let cost = legacy_empty_cost();

    let hash: [u8; 64] = match hash_bytes.as_bytes_ref().as_slice().try_into() {
        Ok(h) => h,
        Err(_) => return Ok(NativeResult::err(cost, INVALID_ECVRF_HASH_LENGTH)),
    };

    let public_key = match bincode::deserialize(public_key_bytes.as_bytes_ref().as_slice()) {
        Ok(pk) => pk,
        Err(_) => return Ok(NativeResult::err(cost, INVALID_ECVRF_PUBLIC_KEY)),
    };

    let proof: ECVRFProof = match bincode::deserialize(proof_bytes.as_bytes_ref().as_slice()) {
        Ok(p) => p,
        Err(_) => return Ok(NativeResult::err(cost, INVALID_ECVRF_PROOF)),
    };

    let result = proof.verify_output(alpha_string.as_bytes_ref().as_slice(), &public_key, &hash);
    Ok(NativeResult::ok(
        cost,
        smallvec![Value::bool(result.is_ok())],
    ))
}
