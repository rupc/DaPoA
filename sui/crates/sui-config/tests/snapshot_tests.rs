// Copyright (c) Mysten Labs, Inc.
// SPDX-License-Identifier: Apache-2.0

// This file contains tests that detect changes in Sui configs.
// If a PR breaks one or more tests here, the PR probably has a real impact
// on a production configuration file. When test failure happens, the PR should
// be marked as a breaking change and reviewers should be aware of this.
//
// Owners and operators of production configuration files can add themselves to
// .github/CODEOWNERS for the corresponding snapshot tests, so they can get notified
// of changes. PRs that modifies snapshot files should wait for reviews from
// code owners (if any) before merging.
//
// To review snapshot changes, and fix snapshot differences,
// 0. Install cargo-insta
// 1. Run `cargo insta test --review` under `./sui-config`.
// 2. Review, accept or reject changes.

use fastcrypto::traits::KeyPair;
use insta::assert_yaml_snapshot;
use multiaddr::Multiaddr;
use rand::rngs::StdRng;
use rand::SeedableRng;
use sui_config::genesis::GenesisChainParameters;
use sui_config::ValidatorInfo;
use sui_config::{genesis::Builder, genesis_config::GenesisConfig};
use sui_protocol_config::ProtocolVersion;
use sui_types::base_types::{ObjectID, SuiAddress};
use sui_types::crypto::{
    generate_proof_of_possession, get_key_pair_from_rng, AccountKeyPair, AuthorityKeyPair,
    AuthorityPublicKeyBytes, NetworkKeyPair,
};

#[test]
#[cfg_attr(msim, ignore)]
fn genesis_config_snapshot_matches() {
    // Test creating fake SuiAddress from PublicKeyBytes.
    let keypair: AuthorityKeyPair = get_key_pair_from_rng(&mut StdRng::from_seed([0; 32])).1;
    let public_key = AuthorityPublicKeyBytes::from(keypair.public());
    let fake_addr = SuiAddress::from(&public_key);

    let fake_obj_id = ObjectID::from(fake_addr);
    let mut genesis_config = GenesisConfig::for_local_testing();
    genesis_config.parameters.timestamp_ms = 0;
    for account in &mut genesis_config.accounts {
        account.address = Some(fake_addr);
        for gas_obj in &mut account.gas_objects {
            gas_obj.object_id = fake_obj_id;
        }
    }
    assert_yaml_snapshot!(genesis_config);
}

#[test]
#[cfg_attr(msim, ignore)]
fn populated_genesis_snapshot_matches() {
    let genesis_config = GenesisConfig::for_local_testing();
    let (_account_keys, objects) = genesis_config
        .generate_accounts(&mut StdRng::from_seed([0; 32]))
        .unwrap();
    let key: AuthorityKeyPair = get_key_pair_from_rng(&mut StdRng::from_seed([0; 32])).1;
    let worker_key: NetworkKeyPair = get_key_pair_from_rng(&mut StdRng::from_seed([0; 32])).1;
    let network_key: NetworkKeyPair = get_key_pair_from_rng(&mut StdRng::from_seed([0; 32])).1;
    let account_key: AccountKeyPair = get_key_pair_from_rng(&mut StdRng::from_seed([0; 32])).1;
    let validator = ValidatorInfo {
        name: "0".into(),
        protocol_key: key.public().into(),
        worker_key: worker_key.public().clone(),
        account_key: account_key.public().clone().into(),
        network_key: network_key.public().clone(),
        gas_price: 1,
        commission_rate: 0,
        network_address: Multiaddr::empty(),
        p2p_address: Multiaddr::empty(),
        narwhal_primary_address: Multiaddr::empty(),
        narwhal_worker_address: Multiaddr::empty(),
        description: String::new(),
        image_url: String::new(),
        project_url: String::new(),
    };
    let pop = generate_proof_of_possession(&key, account_key.public().into());

    let genesis = Builder::new()
        .add_objects(objects)
        .add_validator(validator, pop)
        .with_parameters(GenesisChainParameters {
            timestamp_ms: 10,
            protocol_version: ProtocolVersion::MAX,
            allow_insertion_of_extra_objects: true,
            initial_sui_custody_account_address: SuiAddress::default(),
        })
        .add_validator_signature(&key)
        .build();
    assert_yaml_snapshot!(genesis.validator_set());
    assert_yaml_snapshot!(genesis.sui_system_wrapper_object());
    assert_yaml_snapshot!(genesis.sui_system_object());
    assert_yaml_snapshot!(genesis.clock());
    // Serialized `genesis` is not static and cannot be snapshot tested.
}

#[test]
#[cfg_attr(msim, ignore)]
fn network_config_snapshot_matches() {
    use std::net::{IpAddr, Ipv4Addr, SocketAddr};
    use std::path::PathBuf;
    use sui_config::NetworkConfig;

    let temp_dir = tempfile::tempdir().unwrap();
    let committee_size = 7;
    let rng = StdRng::from_seed([0; 32]);
    let mut network_config = NetworkConfig::generate_with_rng(temp_dir.path(), committee_size, rng);
    // TODO: Inject static temp path and port numbers, instead of clearing them.
    for mut validator_config in &mut network_config.validator_configs {
        validator_config.db_path = PathBuf::from("/tmp/foo/");
        validator_config.network_address = Multiaddr::empty();
        let fake_socket = SocketAddr::new(IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)), 1);
        validator_config.json_rpc_address = fake_socket;
        validator_config.metrics_address = fake_socket;
        validator_config.p2p_config.listen_address = fake_socket;
        validator_config.p2p_config.external_address = None;
        validator_config.admin_interface_port = 8888;
        let metrics_addr: Multiaddr = "/ip4/127.0.0.1/tcp/1234".parse().unwrap();
        let primary_network_admin_server_port = 5678;
        let worker_network_admin_server_base_port = 8765;
        if let Some(consensus_config) = validator_config.consensus_config.as_mut() {
            consensus_config.address = Multiaddr::empty();
            consensus_config.db_path = PathBuf::from("/tmp/foo/");
            consensus_config.internal_worker_address = Some(Multiaddr::empty());
            consensus_config
                .narwhal_config
                .consensus_api_grpc
                .socket_addr = Multiaddr::empty();
            consensus_config
                .narwhal_config
                .prometheus_metrics
                .socket_addr = metrics_addr;
            consensus_config
                .narwhal_config
                .network_admin_server
                .primary_network_admin_server_port = primary_network_admin_server_port;
            consensus_config
                .narwhal_config
                .network_admin_server
                .worker_network_admin_server_base_port = worker_network_admin_server_base_port;
        }
    }
    assert_yaml_snapshot!(network_config, {
        ".genesis" => "[fake genesis]",
        ".validator_configs[].genesis.genesis" => "[fake genesis]",
    });
}
