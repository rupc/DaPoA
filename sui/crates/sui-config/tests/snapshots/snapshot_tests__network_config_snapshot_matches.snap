---
source: crates/sui-config/tests/snapshot_tests.rs
expression: network_config
---
validator_configs:
  - protocol-key-pair:
      value: VTDx4HjVmRBqdqBWg2zN+zcFE20io3CrBchGy/iV1lo=
    worker-key-pair:
      value: AH7M/Ot6iUd/Jj47r5aGmQROL24mxT4K8EF1Gvjhk0zT
    account-key-pair:
      value: AGXlMKrYkj+CLojtVPwa9rE1QUKZHJA+XGhfgWp7porR
    network-key-pair:
      value: AClK2/gtiS+cypPl2il6YC49PJgkHoAgnpKZ0dEiwIUw
    db-path: /tmp/foo/
    network-address: ""
    json-rpc-address: "0.0.0.0:1"
    metrics-address: "0.0.0.0:1"
    admin-interface-port: 8888
    consensus-config:
      address: ""
      db-path: /tmp/foo/
      internal-worker-address: ""
      timeout-secs: 60
      narwhal-config:
        header_num_of_batches_threshold: 32
        max_header_num_of_batches: 1000
        max_header_delay: 2000ms
        min_header_delay: 500ms
        gc_depth: 50
        sync_retry_delay: 5000ms
        sync_retry_nodes: 3
        batch_size: 500000
        max_batch_delay: 100ms
        block_synchronizer:
          range_synchronize_timeout: 30000ms
          certificates_synchronize_timeout: 30000ms
          payload_synchronize_timeout: 30000ms
          payload_availability_timeout: 30000ms
          handler_certificate_deliver_timeout: 30000ms
        consensus_api_grpc:
          socket_addr: ""
          get_collections_timeout: 5000ms
          remove_collections_timeout: 5000ms
        max_concurrent_requests: 500000
        prometheus_metrics:
          socket_addr: /ip4/127.0.0.1/tcp/1234
        network_admin_server:
          primary_network_admin_server_port: 5678
          worker_network_admin_server_base_port: 8765
        anemo:
          send_certificate_rate_limit: ~
          get_payload_availability_rate_limit: ~
          get_certificates_rate_limit: ~
          report_batch_rate_limit: ~
          request_batch_rate_limit: ~
    enable-event-processing: false
    epoch-duration-ms: 86400000
    grpc-load-shed: ~
    grpc-concurrency-limit: 20000000000
    p2p-config:
      listen-address: "0.0.0.0:1"
    genesis:
      genesis: "[fake genesis]"
    authority-store-pruning-config:
      objects-num-latest-versions-to-retain: 2
      objects-pruning-period-secs: 86400
      objects-pruning-initial-delay-secs: 3600
      num-latest-epoch-dbs-to-retain: 3
      epoch-db-pruning-period-secs: 3600
      num-epochs-to-retain: 18446744073709551615
    end-of-epoch-broadcast-channel-capacity: 128
    checkpoint-executor-config:
      checkpoint-execution-max-concurrency: 200
      local-execution-timeout-sec: 10
  - protocol-key-pair:
      value: avYcyVgYMXTyaUYh9IRwLK0gSzl7YF6ZQDAbrS1Bhvo=
    worker-key-pair:
      value: AKTwSrpgq5BEhJgIhv+1H0jMiJQMigkGsrAW9EQPaMSI
    account-key-pair:
      value: AHs3lfh8LlhanWE/zTjOqDFkNiDL2ouxjjJ+WDT5/hVg
    network-key-pair:
      value: ANuTFymJGw6l1rSGV5du4G4ROOLd4adUItxxVhy31lnN
    db-path: /tmp/foo/
    network-address: ""
    json-rpc-address: "0.0.0.0:1"
    metrics-address: "0.0.0.0:1"
    admin-interface-port: 8888
    consensus-config:
      address: ""
      db-path: /tmp/foo/
      internal-worker-address: ""
      timeout-secs: 60
      narwhal-config:
        header_num_of_batches_threshold: 32
        max_header_num_of_batches: 1000
        max_header_delay: 2000ms
        min_header_delay: 500ms
        gc_depth: 50
        sync_retry_delay: 5000ms
        sync_retry_nodes: 3
        batch_size: 500000
        max_batch_delay: 100ms
        block_synchronizer:
          range_synchronize_timeout: 30000ms
          certificates_synchronize_timeout: 30000ms
          payload_synchronize_timeout: 30000ms
          payload_availability_timeout: 30000ms
          handler_certificate_deliver_timeout: 30000ms
        consensus_api_grpc:
          socket_addr: ""
          get_collections_timeout: 5000ms
          remove_collections_timeout: 5000ms
        max_concurrent_requests: 500000
        prometheus_metrics:
          socket_addr: /ip4/127.0.0.1/tcp/1234
        network_admin_server:
          primary_network_admin_server_port: 5678
          worker_network_admin_server_base_port: 8765
        anemo:
          send_certificate_rate_limit: ~
          get_payload_availability_rate_limit: ~
          get_certificates_rate_limit: ~
          report_batch_rate_limit: ~
          request_batch_rate_limit: ~
    enable-event-processing: false
    epoch-duration-ms: 86400000
    grpc-load-shed: ~
    grpc-concurrency-limit: 20000000000
    p2p-config:
      listen-address: "0.0.0.0:1"
    genesis:
      genesis: "[fake genesis]"
    authority-store-pruning-config:
      objects-num-latest-versions-to-retain: 2
      objects-pruning-period-secs: 86400
      objects-pruning-initial-delay-secs: 3600
      num-latest-epoch-dbs-to-retain: 3
      epoch-db-pruning-period-secs: 3600
      num-epochs-to-retain: 18446744073709551615
    end-of-epoch-broadcast-channel-capacity: 128
    checkpoint-executor-config:
      checkpoint-execution-max-concurrency: 200
      local-execution-timeout-sec: 10
  - protocol-key-pair:
      value: OXnx3yM1C/ppgnDMx/o1d49fJs7E05kq11mXNae/O+I=
    worker-key-pair:
      value: ANQ4duxTsOdofUAHfSIo8kauNj7SRmLZgmdkb12DwxF6
    account-key-pair:
      value: AGGCaIy1OwvvLtr2rcHewlJJBILLzhrEao4QmF0D2UTz
    network-key-pair:
      value: AKFIgWPoTZb9/zUu/1P8wFtxDDYuObk1xr/QaxZ2H48i
    db-path: /tmp/foo/
    network-address: ""
    json-rpc-address: "0.0.0.0:1"
    metrics-address: "0.0.0.0:1"
    admin-interface-port: 8888
    consensus-config:
      address: ""
      db-path: /tmp/foo/
      internal-worker-address: ""
      timeout-secs: 60
      narwhal-config:
        header_num_of_batches_threshold: 32
        max_header_num_of_batches: 1000
        max_header_delay: 2000ms
        min_header_delay: 500ms
        gc_depth: 50
        sync_retry_delay: 5000ms
        sync_retry_nodes: 3
        batch_size: 500000
        max_batch_delay: 100ms
        block_synchronizer:
          range_synchronize_timeout: 30000ms
          certificates_synchronize_timeout: 30000ms
          payload_synchronize_timeout: 30000ms
          payload_availability_timeout: 30000ms
          handler_certificate_deliver_timeout: 30000ms
        consensus_api_grpc:
          socket_addr: ""
          get_collections_timeout: 5000ms
          remove_collections_timeout: 5000ms
        max_concurrent_requests: 500000
        prometheus_metrics:
          socket_addr: /ip4/127.0.0.1/tcp/1234
        network_admin_server:
          primary_network_admin_server_port: 5678
          worker_network_admin_server_base_port: 8765
        anemo:
          send_certificate_rate_limit: ~
          get_payload_availability_rate_limit: ~
          get_certificates_rate_limit: ~
          report_batch_rate_limit: ~
          request_batch_rate_limit: ~
    enable-event-processing: false
    epoch-duration-ms: 86400000
    grpc-load-shed: ~
    grpc-concurrency-limit: 20000000000
    p2p-config:
      listen-address: "0.0.0.0:1"
    genesis:
      genesis: "[fake genesis]"
    authority-store-pruning-config:
      objects-num-latest-versions-to-retain: 2
      objects-pruning-period-secs: 86400
      objects-pruning-initial-delay-secs: 3600
      num-latest-epoch-dbs-to-retain: 3
      epoch-db-pruning-period-secs: 3600
      num-epochs-to-retain: 18446744073709551615
    end-of-epoch-broadcast-channel-capacity: 128
    checkpoint-executor-config:
      checkpoint-execution-max-concurrency: 200
      local-execution-timeout-sec: 10
  - protocol-key-pair:
      value: CyNkjqNVr3HrHTH7f/NLs7u5lUHJzuPAw0PqMTD2y2s=
    worker-key-pair:
      value: ABB1+x4nRFoyYG3rKucsVmnAZB1Mf8ACwfdDJvEomhGW
    account-key-pair:
      value: AJFy2uQrEISFM5YRQRBUz9XisSepQRkWMzZDo6w2ppVP
    network-key-pair:
      value: AI0OYrH3grOXDvJEH0aDYhOhMdudJ4hLVwh+1OGNyoXe
    db-path: /tmp/foo/
    network-address: ""
    json-rpc-address: "0.0.0.0:1"
    metrics-address: "0.0.0.0:1"
    admin-interface-port: 8888
    consensus-config:
      address: ""
      db-path: /tmp/foo/
      internal-worker-address: ""
      timeout-secs: 60
      narwhal-config:
        header_num_of_batches_threshold: 32
        max_header_num_of_batches: 1000
        max_header_delay: 2000ms
        min_header_delay: 500ms
        gc_depth: 50
        sync_retry_delay: 5000ms
        sync_retry_nodes: 3
        batch_size: 500000
        max_batch_delay: 100ms
        block_synchronizer:
          range_synchronize_timeout: 30000ms
          certificates_synchronize_timeout: 30000ms
          payload_synchronize_timeout: 30000ms
          payload_availability_timeout: 30000ms
          handler_certificate_deliver_timeout: 30000ms
        consensus_api_grpc:
          socket_addr: ""
          get_collections_timeout: 5000ms
          remove_collections_timeout: 5000ms
        max_concurrent_requests: 500000
        prometheus_metrics:
          socket_addr: /ip4/127.0.0.1/tcp/1234
        network_admin_server:
          primary_network_admin_server_port: 5678
          worker_network_admin_server_base_port: 8765
        anemo:
          send_certificate_rate_limit: ~
          get_payload_availability_rate_limit: ~
          get_certificates_rate_limit: ~
          report_batch_rate_limit: ~
          request_batch_rate_limit: ~
    enable-event-processing: false
    epoch-duration-ms: 86400000
    grpc-load-shed: ~
    grpc-concurrency-limit: 20000000000
    p2p-config:
      listen-address: "0.0.0.0:1"
    genesis:
      genesis: "[fake genesis]"
    authority-store-pruning-config:
      objects-num-latest-versions-to-retain: 2
      objects-pruning-period-secs: 86400
      objects-pruning-initial-delay-secs: 3600
      num-latest-epoch-dbs-to-retain: 3
      epoch-db-pruning-period-secs: 3600
      num-epochs-to-retain: 18446744073709551615
    end-of-epoch-broadcast-channel-capacity: 128
    checkpoint-executor-config:
      checkpoint-execution-max-concurrency: 200
      local-execution-timeout-sec: 10
  - protocol-key-pair:
      value: X/I/kM+KvHcxAKEf2UU6Sr7SpN3bhiE9nP5CuM/iIY0=
    worker-key-pair:
      value: AFJKmNZNkqyS85XUqAKsw2adb7nVejMwGaqI2ptF2NmT
    account-key-pair:
      value: ACdIQ0IILpTytmU6RjVq3Ifr4O9zkPZVn0V7vbjfkKPP
    network-key-pair:
      value: AFKq0My7gQHTDG58VQMYImhU18+j2le9HtF08nhehbvK
    db-path: /tmp/foo/
    network-address: ""
    json-rpc-address: "0.0.0.0:1"
    metrics-address: "0.0.0.0:1"
    admin-interface-port: 8888
    consensus-config:
      address: ""
      db-path: /tmp/foo/
      internal-worker-address: ""
      timeout-secs: 60
      narwhal-config:
        header_num_of_batches_threshold: 32
        max_header_num_of_batches: 1000
        max_header_delay: 2000ms
        min_header_delay: 500ms
        gc_depth: 50
        sync_retry_delay: 5000ms
        sync_retry_nodes: 3
        batch_size: 500000
        max_batch_delay: 100ms
        block_synchronizer:
          range_synchronize_timeout: 30000ms
          certificates_synchronize_timeout: 30000ms
          payload_synchronize_timeout: 30000ms
          payload_availability_timeout: 30000ms
          handler_certificate_deliver_timeout: 30000ms
        consensus_api_grpc:
          socket_addr: ""
          get_collections_timeout: 5000ms
          remove_collections_timeout: 5000ms
        max_concurrent_requests: 500000
        prometheus_metrics:
          socket_addr: /ip4/127.0.0.1/tcp/1234
        network_admin_server:
          primary_network_admin_server_port: 5678
          worker_network_admin_server_base_port: 8765
        anemo:
          send_certificate_rate_limit: ~
          get_payload_availability_rate_limit: ~
          get_certificates_rate_limit: ~
          report_batch_rate_limit: ~
          request_batch_rate_limit: ~
    enable-event-processing: false
    epoch-duration-ms: 86400000
    grpc-load-shed: ~
    grpc-concurrency-limit: 20000000000
    p2p-config:
      listen-address: "0.0.0.0:1"
    genesis:
      genesis: "[fake genesis]"
    authority-store-pruning-config:
      objects-num-latest-versions-to-retain: 2
      objects-pruning-period-secs: 86400
      objects-pruning-initial-delay-secs: 3600
      num-latest-epoch-dbs-to-retain: 3
      epoch-db-pruning-period-secs: 3600
      num-epochs-to-retain: 18446744073709551615
    end-of-epoch-broadcast-channel-capacity: 128
    checkpoint-executor-config:
      checkpoint-execution-max-concurrency: 200
      local-execution-timeout-sec: 10
  - protocol-key-pair:
      value: N272EiFDyKtxRbDKbyN6ujenJ+skPcRoc/XolpOLGnU=
    worker-key-pair:
      value: AB5aMuKZ3/KVhBxj/uDhbF8+emy3SCYZA9jvjnSjKggO
    account-key-pair:
      value: AKbzEnI6DEfw2jdDOSDsUtlaRgufTepL+oA5ut7ny0Oc
    network-key-pair:
      value: ALT7t4M6FqBWx257mlKdcIVihVe5rQ1XzrUxw1lkdmaF
    db-path: /tmp/foo/
    network-address: ""
    json-rpc-address: "0.0.0.0:1"
    metrics-address: "0.0.0.0:1"
    admin-interface-port: 8888
    consensus-config:
      address: ""
      db-path: /tmp/foo/
      internal-worker-address: ""
      timeout-secs: 60
      narwhal-config:
        header_num_of_batches_threshold: 32
        max_header_num_of_batches: 1000
        max_header_delay: 2000ms
        min_header_delay: 500ms
        gc_depth: 50
        sync_retry_delay: 5000ms
        sync_retry_nodes: 3
        batch_size: 500000
        max_batch_delay: 100ms
        block_synchronizer:
          range_synchronize_timeout: 30000ms
          certificates_synchronize_timeout: 30000ms
          payload_synchronize_timeout: 30000ms
          payload_availability_timeout: 30000ms
          handler_certificate_deliver_timeout: 30000ms
        consensus_api_grpc:
          socket_addr: ""
          get_collections_timeout: 5000ms
          remove_collections_timeout: 5000ms
        max_concurrent_requests: 500000
        prometheus_metrics:
          socket_addr: /ip4/127.0.0.1/tcp/1234
        network_admin_server:
          primary_network_admin_server_port: 5678
          worker_network_admin_server_base_port: 8765
        anemo:
          send_certificate_rate_limit: ~
          get_payload_availability_rate_limit: ~
          get_certificates_rate_limit: ~
          report_batch_rate_limit: ~
          request_batch_rate_limit: ~
    enable-event-processing: false
    epoch-duration-ms: 86400000
    grpc-load-shed: ~
    grpc-concurrency-limit: 20000000000
    p2p-config:
      listen-address: "0.0.0.0:1"
    genesis:
      genesis: "[fake genesis]"
    authority-store-pruning-config:
      objects-num-latest-versions-to-retain: 2
      objects-pruning-period-secs: 86400
      objects-pruning-initial-delay-secs: 3600
      num-latest-epoch-dbs-to-retain: 3
      epoch-db-pruning-period-secs: 3600
      num-epochs-to-retain: 18446744073709551615
    end-of-epoch-broadcast-channel-capacity: 128
    checkpoint-executor-config:
      checkpoint-execution-max-concurrency: 200
      local-execution-timeout-sec: 10
  - protocol-key-pair:
      value: a74f03IOjL8ZFSWFChFVEi+wiMwHNwNCPDGIYkGfgjs=
    worker-key-pair:
      value: AOUVpWBdto0vYu+sy5fIz9g9w/BguYbfiD9XVlARmcYv
    account-key-pair:
      value: AEORJLwnMuXMNSSkrcErNL/0AcltCPCVwqnZ2oKLqQjI
    network-key-pair:
      value: APhBCF+ITkAe/cag6/AqdFBjTQuLBBPOPfzkUxmNYsMX
    db-path: /tmp/foo/
    network-address: ""
    json-rpc-address: "0.0.0.0:1"
    metrics-address: "0.0.0.0:1"
    admin-interface-port: 8888
    consensus-config:
      address: ""
      db-path: /tmp/foo/
      internal-worker-address: ""
      timeout-secs: 60
      narwhal-config:
        header_num_of_batches_threshold: 32
        max_header_num_of_batches: 1000
        max_header_delay: 2000ms
        min_header_delay: 500ms
        gc_depth: 50
        sync_retry_delay: 5000ms
        sync_retry_nodes: 3
        batch_size: 500000
        max_batch_delay: 100ms
        block_synchronizer:
          range_synchronize_timeout: 30000ms
          certificates_synchronize_timeout: 30000ms
          payload_synchronize_timeout: 30000ms
          payload_availability_timeout: 30000ms
          handler_certificate_deliver_timeout: 30000ms
        consensus_api_grpc:
          socket_addr: ""
          get_collections_timeout: 5000ms
          remove_collections_timeout: 5000ms
        max_concurrent_requests: 500000
        prometheus_metrics:
          socket_addr: /ip4/127.0.0.1/tcp/1234
        network_admin_server:
          primary_network_admin_server_port: 5678
          worker_network_admin_server_base_port: 8765
        anemo:
          send_certificate_rate_limit: ~
          get_payload_availability_rate_limit: ~
          get_certificates_rate_limit: ~
          report_batch_rate_limit: ~
          request_batch_rate_limit: ~
    enable-event-processing: false
    epoch-duration-ms: 86400000
    grpc-load-shed: ~
    grpc-concurrency-limit: 20000000000
    p2p-config:
      listen-address: "0.0.0.0:1"
    genesis:
      genesis: "[fake genesis]"
    authority-store-pruning-config:
      objects-num-latest-versions-to-retain: 2
      objects-pruning-period-secs: 86400
      objects-pruning-initial-delay-secs: 3600
      num-latest-epoch-dbs-to-retain: 3
      epoch-db-pruning-period-secs: 3600
      num-epochs-to-retain: 18446744073709551615
    end-of-epoch-broadcast-channel-capacity: 128
    checkpoint-executor-config:
      checkpoint-execution-max-concurrency: 200
      local-execution-timeout-sec: 10
account_keys:
  - 10wECHkYvXqL5/CY6WhjbfFPotZb5tjEbpmumqbRxuk=
  - ZTWBfKEmFOyYM9oBU9dNfREBuAU5fm2OBhg/vPtI00c=
  - +0LdRe35y6TgXPz6qcArpf0T8/Hh7zkjJL0yT19jbjc=
  - cHifntmjxd4QLaC71WRMoOeCpvicyDJMbTKRuo2v3R8=
  - mTzV/JVz4RdheOnQVFF3xuCPFF4AYAPyShHQCUizJX8=
genesis: "[fake genesis]"

